internal/config/common.go

```diff
@@ -50,35 +53,7 @@ func GetConfiguration[T any](filename string, defaultGenerator func() T) (T, err
 	var config T
 
 	if _, err := toml.DecodeFile(configPath, &config); err != nil {
-		return defaultGenerator(), nil
-	}
-	return config, nil
-}
-
-func CreateTempConfigFile(content string, t *testing.T) (string, func()) {
-	t.Helper()
-
-	exePath, err := os.Executable()
-
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	tmpfile, err := os.CreateTemp(filepath.Dir(exePath), "config.toml")
-
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if _, err := tmpfile.Write([]byte(content)); err != nil {
-		t.Fatal(err)
-	}
-
-	if err := tmpfile.Close(); err != nil {
-		t.Fatal(err)
-	}
-
-	return tmpfile.Name(), func() {
-		os.Remove(tmpfile.Name())
+		return defaultGenerator(), err
 	}
+	return config.Merge(defaultGenerator()).(T), nil
```

Comment:
You can avoid needing to implement merging, due to how decoding goes into an existing struct: https://go.dev/play/p/dx-RXKitbbk

Reply:
This seems helpful. Would mean we can remove the merge-related code.